include/envoy/network/listener.h

```diff
@@ -43,6 +43,14 @@ class ListenerConfig {
    */
   virtual bool bindToPort() PURE;
 
+  /**
+   * @return bool if a connection should be handed off to another Listener after the original
+   *         destination address has been restored. 'true' when 'use_original_dst' flag in listener
+   *         configuration is set, false otherwise. Note that this flag will be removed from the v2
+   *         API.
+   */
+  virtual bool handOffRestoredDestinations() const PURE;
```

Comment:
Should this be not plural? `handOffRestoredDestination()` ? And same with all variables similarly named?

Reply:
A listener accepts many connections, each of which has a destination address, and the listener will either hand all restored destinations off to another listener (if found) or not. To me this calls for a plural, but I'm happy to change it you don't buy this line of reasoning.

Reply:
OK makes sense. I guess I would probably name it `handOffRestoredDestinationConnections()`, but I don't feel that strongly about it.

Reply:
OK, I'll change this, hold on for a while :-)